reactor/src/main/java/com/iluwatar/reactor/app/LoggingHandler.java

```diff
@@ -0,0 +1,60 @@
+package com.iluwatar.reactor.app;
+
+import java.nio.ByteBuffer;
+import java.nio.channels.SelectionKey;
+
+import com.iluwatar.reactor.framework.AbstractNioChannel;
+import com.iluwatar.reactor.framework.ChannelHandler;
+import com.iluwatar.reactor.framework.NioDatagramChannel.DatagramPacket;
+
+/**
+ * Logging server application logic. It logs the incoming requests on standard console and returns
+ * a canned acknowledgement back to the remote peer.
+ * 
+ * @author npathai
+ */
+public class LoggingHandler implements ChannelHandler {
+
+	private static final byte[] ACK = "Data logged successfully".getBytes();
+
+	/**
+	 * Decodes the received data and logs it on standard console.
+	 */
+	@Override
+	public void handleChannelRead(AbstractNioChannel channel, Object readObject, SelectionKey key) {
+		/*
+		 * As this handler is attached with both TCP and UDP channels we need to check whether
+		 * the data received is a ByteBuffer (from TCP channel) or a DatagramPacket (from UDP channel).  
+		 */
```

Comment:
Assuming you are taking about `instanceof` checks, we can have different subclasses of `LoggingHandler`, one specific to TCP and other to UDP. The main business logic of logging can be kept in base class.